src/main/java/org/sigmah/client/computation/ComputationTriggerManager.java

```diff
@@ -0,0 +1,263 @@
+	 */
+	public void prepareForProject(final ProjectDTO project) {
+        
+		this.container = project;
+        
+		clearMaps();
+        
+		final ProjectModelDTO model = project.getProjectModel();
+		
+		for (final ProjectDTO.LocalizedElement<ComputationElementDTO> localizedElement : project.getLocalizedElements(ComputationElementDTO.class)) {
+			final ComputationElementDTO computationElement = localizedElement.getElement();
+			prepareForComputationElement(computationElement, model);
+		}
+	}
+    
+    /**
+	 * Prepare the trigger manager.
+	 * 
+	 * @param orgUnit
+     *          OrgUnit to display.
+	 */
+    public void prepareForOrgUnit(final OrgUnitDTO orgUnit) {
+        
+        this.container = orgUnit;
+        
+		clearMaps();
+        
+        final OrgUnitModelDTO model = orgUnit.getOrgUnitModel();
+        
+        for (final OrgUnitDTO.LocalizedElement localizedElement : orgUnit.getLocalizedElements(ComputationElementDTO.class)) {
+			final ComputationElementDTO computationElement = (ComputationElementDTO) localizedElement.getElement();
+			prepareForComputationElement(computationElement, model);
+		}
+    }
+
+    /**
+     * Remove the content of every maps.
+     */
+    private void clearMaps() {
+        this.dependencies.clear();
+        this.computations.clear();
+        this.components.clear();
+        this.elementsWithHandlers.clear();
+    }
+
+    /**
+     * List the dependencies of the given element.
+     * 
+     * @param computationElement
+     *          Computation element to prepare.
+     * @param model
+     *          Model of the current container.
+     */
+    private void prepareForComputationElement(final ComputationElementDTO computationElement, final IsModel model) {
+        final Computation computation = computationElement.getComputationForModel(model);
+        computations.put(computationElement, computation);
+        
+        for (final FlexibleElementDTO dependency : computation.getDependencies()) {
+            List<ComputationElementDTO> list = dependencies.get(dependency);
+            
+            if (list == null) {
+                list = new ArrayList<ComputationElementDTO>();
+                dependencies.put(dependency, list);
+            }
+            
+            list.add(computationElement);
+        }
+    }
+    
+	/**
+	 * Add a value change handler to the given element if it is a dependency
+	 * of a computation.
+	 * 
+	 * @param element
+     *          Element to listen.
+	 * @param component
+     *          Component associated to the given element.
+	 * @param modifications
+     *          Value change list.
+	 */
+	public void listenToValueChangesOfElement(final FlexibleElementDTO element, final Component component, final List<ValueEvent> modifications) {
+		
+		if (element instanceof ComputationElementDTO) {
+			components.put(element, (Field<String>) component);
+            elementsWithHandlers.put(element.getId(), (ComputationElementDTO) element);
```

Comment:
Oops, it seems my formatter was lost when switching between branches. The main indentation style used on Sigmah is tab. Should I change this file indentation to all-tabs ?